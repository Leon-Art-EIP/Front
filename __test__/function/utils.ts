import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
