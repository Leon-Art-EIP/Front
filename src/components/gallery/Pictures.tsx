interface IPicturesProps {
  pictures: string[];
}

/* c8 ignore start */
export default function Pictures(props: IPicturesProps): JSX.Element {
  return (
    <div className="grid grid-cols-4 gap-8 place-items-center justify-center py-4 px-4">
      {props.pictures.map((url, index) => (
        <div key={index} className="w-55 h-55 rounded-xl overflow-hidden bg-slate-300">
          <img src={url} alt="art" className="w-full h-full object-cover" />
        </div>
      ))}
    </div>
  );
}
/* c8 ignore stop */
