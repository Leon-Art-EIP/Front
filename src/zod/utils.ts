export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
