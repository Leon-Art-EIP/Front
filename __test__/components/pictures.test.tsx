import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Pictures from "../../src/components/gallery/Pictures";

test("Pictures empty snapshot", () => {
  testSnapshot(<Pictures pictures={[]} />);
});

test("Pictures snapshot", () => {
  testSnapshot(<Pictures pictures={["test1", "test2", "test3"]} />);
});
