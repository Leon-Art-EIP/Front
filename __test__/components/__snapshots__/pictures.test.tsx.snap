// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pictures empty snapshot 1`] = `
<div
  className="bg-background grid grid-cols-4 gap-8 place-items-center justify-center py-4 px-4"
/>
`;

exports[`Pictures snapshot 1`] = `
<div
  className="bg-background grid grid-cols-4 gap-8 place-items-center justify-center py-4 px-4"
>
  <div
    className="w-full aspect-square rounded-xl overflow-hidden"
  >
    <img
      alt="art"
      className="w-full h-full object-cover"
      src="test1"
    />
  </div>
  <div
    className="w-full aspect-square rounded-xl overflow-hidden"
  >
    <img
      alt="art"
      className="w-full h-full object-cover"
      src="test2"
    />
  </div>
  <div
    className="w-full aspect-square rounded-xl overflow-hidden"
  >
    <img
      alt="art"
      className="w-full h-full object-cover"
      src="test3"
    />
  </div>
</div>
`;
